source/module_base/test/complexarray_test.cpp

```diff
@@ -0,0 +1,207 @@
+*       - assign a ComplexArray to another a ComplexArray
+*
+*   - operator "+":
+*       - one ComplexArray plus another ComplexArray that has
+*         the same dimension.
+*       - throw error when one ComplexArray plus another ComplexArray 
+*         that has different dimension.
+*
+*   - operator "+=":
+*       - one ComplexArray plus another ComplexArray, and assign the 
+*         value to first ComplexArray
+*       - throw error when one ComplexArray plus another ComplexArray
+*         that has different dimension.
+*
+*   - operator "-":
+*       - one ComplexArray minus another ComplexArray that has
+*         the same dimension.
+*       - throw error when one ComplexArray minus another ComplexArray
+*         that has different dimension.
+*
+*   - operator "-=":
+*       - one ComplexArray minus another ComplexArray, and assign the
+*         value to first ComplexArray
+*       - throw error when one ComplexArray minus another ComplexArray
+*         that has different dimension.
+*
+*   - operator "*":
+*       - one ComplexArray is multiplied by a double
+*       - one ComplexArray is multiplied by a complex
+*       - one ComplexArray is multiplied by another ComplexArray that has same dimension
+*       - throw error when one ComplexArray is miltiplied by another ComplexArray
+*         that has different dimension.
+*
+*   - operator "*=":
+*       similar as "*"
+*
+*   - oprator "()":
+*       - assess the element
+*/
+
+
+class ComplexArray_test : public testing::Test
+{
+    protected:
+        ModuleBase::ComplexArray a2,a4,b2,b4,c2,c4,d2;
+        ModuleBase::ComplexArray a2_plus_b2,a2_minus_b2,a2_mult_b2,a2_mult_3;
+        std::complex<double> com1 {1.0,2.0};
+        std::complex<double> com2 {3.0,4.0};
+
+        void SetUp()
+        {
+            a2 = ModuleBase::ComplexArray(2,1,1,1);  // if define this class as a matrix
+            b2 = ModuleBase::ComplexArray(2,1,1,1);  // of 4 dimesions,
+            c2 = ModuleBase::ComplexArray(2,1,1,1);  // is a2 +/-/* d2 allowed or not???
+            d2 = ModuleBase::ComplexArray(1,2,1,1);  // it does not matter, this situation will not appear in ABACUS
+            a2_plus_b2 = ModuleBase::ComplexArray(2,1,1,1);
+            a2_minus_b2 = ModuleBase::ComplexArray(2,1,1,1);
+            a2_mult_b2 = ModuleBase::ComplexArray(2,1,1,1);
+            a2_mult_3 = ModuleBase::ComplexArray(2,1,1,1);
+            a4 = ModuleBase::ComplexArray(2,2,1,1);
+            b4 = ModuleBase::ComplexArray(2,2,1,1);
+            c4 = ModuleBase::ComplexArray(2,2,1,1);
+            a2 = com1;
+            b2 = com2;
+            a4 = com1;
+            b4 = com2;
+            d2 = com1;
+            a2_plus_b2 = com1 + com2;
+            a2_minus_b2 = com1 - com2;
+            a2_mult_b2 = com1 * com2;
+            a2_mult_3 = com1 * 3.0;
+        }
+
+};
+
+bool Compare(ModuleBase::ComplexArray &ca1, ModuleBase::ComplexArray &ca2)
```

Comment:
Consider adding this function to `ComplexArray` class as `operator ==`?

Reply:
> good idea, I can try to add this function